lib/MFA.php

```diff
@@ -79,7 +79,7 @@ public function challengeFactor(
         $authenticationFactorId,
         $smsTemplate = null
     ) {
-        $challengePath = "auth/factors/challenge";
+        $challengePath = "auth/factors/{$authenticationFactorId}/challenge";
 
         if (!isset($authenticationFactorId)) {
```

Comment:
Should we move this conditional up so that we can check it before building the URL?